thaomonster/ACNH-Collections
# Feature/favoriting

### What does this PR do? 
>Clearly and concisely describe the feature. 

This PR adds favorite functionality and appears on the page. It also have simple error messages.

### Is this a feature or Fix/Refactor?  
Feature

### Problems Encountered & Resulting Solutions  
>State any problems encountered 
>Describe your code changes in detail for reviewers.

Originally had a new route for favorites but had an issue with cors. Instead it is set up similar to how search functionality is working were it switches list.

### Where should the reviewer start?
>What file(s) and line(s) are where changes are made?
`src/Components/App/App.js ` lines: 11-13, 26-31, 36-41, 43-48, 50-57
`src/Components/Header/Header.js` lines: 3, 5, 14-17, 20-38
`src/Components/ProfilePage/ProfilePage.js ` lines: 12-21

### How should this be tested?
>What steps need to be taken to confirm functionality? 
1. `git fetch` branch
2. run `npm start`
3. Click on a villager
4. Click on the blue leaf icon in the passport and it should turn green now.
5. Click on the leaf in the header and it will take you to the favorite list
6. Click on the title to go back home.

### Steps To Take
>Any questions and possible steps moving forward.
Try to make it more user friendly.

### This PR is related to which issues:
>Tag issue here: `Closes #<issue_number>`
Closes #9 and Closes #17
